File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<string,vector<string>> map;

        for(auto str:strs){
           string data = str;
           std::sort(data.begin(), data.end());
           map[data].push_back(str);
        }

        vector<vector<string>> res;
        for(auto[k,v]: map){
            res.push_back(v);
        }
        return res;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        map = {}
        

        for str in strs:
            data = sorted(str)
            found = ''.join(data)

            if found not in map:
                map[found] = []
            map[found].append(str)

        print(map)
        return list(map.values())
